### PR TITLE
MM-12496 Remove flux store usages in team* components

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Remove flux store usages in team* components and some other minor clean-up.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12496

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by jwilander** [comment]
@lindy65 could you smoke test the team sidebar, team settings and team manage members modal?

**Comment by lindy65** [comment]
Thanks @jwilander 

Tested team sidebar, team settings and team add/manage members modals

1. I get a JS console error when trying to add a member to the team and the member is not added to the team

image

2. When joining the team from a team invite link, I get the following JS console error (but am able to join the team)

image

3. Not sure whether related to this PR but tested adding members of the team to channels and opening DM/GM messages and get this JS console error for both actions

image

**Comment by jwilander** [comment]
Thanks @lindy65, those are all unrelated issues that are on master and we have tickets to fix those already

**Comment by lindy65** [comment]
Thanks @jwilander - looks good to me then 👍

**Comment by cpanato** [comment]
@jwilander need rebase

File: components/team_sidebar/selectors.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+import {getJoinableTeams, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+
+export const moreTeamsToJoin = createSelector(
```
Comment (saturninoabril) [inline components/team_sidebar/selectors.js]: Sorry for late review.
Looks like we could use this selector to SidebarRightMenu and SidebarHeaderDropdown as well.

**Pull request merged**